packages/abi-gen/package.json

```diff
@@ -8,16 +8,33 @@
     "main": "lib/src/index.js",
     "types": "lib/src/index.d.ts",
     "scripts": {
-        "lint": "tslint --format stylish --project .",
-        "fix": "tslint --fix --format stylish --project .",
-        "clean": "shx rm -rf lib",
+        "lint": "tslint --format stylish --project . --exclude ./test/generate_test/generated-wrappers/**/* --exclude ./test/generate_test/generated-artifacts/**/* --exclude **/lib/**/* && yarn lint-contracts",
+        "fix": "tslint --fix --format stylish --project . --exclude ./test/generate_test/generated-wrappers/**/* --exclude ./test/generate_test/generated-artifacts/**/* --exclude **/lib/**/* && yarn lint-contracts",
+        "clean": "shx rm -rf lib test/generate_test/generated-artifacts test/generate_test/generated-wrappers",
         "build": "tsc -b",
+        "pre_test": "run-s compile generate_contract_wrappers build:test",
+        "build:test": "tsc --project test/generate_test/tsconfig.json ",
         "build:ci": "yarn build",
-        "test": "yarn run_mocha",
-        "run_mocha": "mocha --require source-map-support/register --require make-promises-safe lib/test/**/*_test.js --bail --exit",
+        "test": "yarn pre_test && yarn run_mocha",
+        "run_mocha": "mocha --require source-map-support/register --require make-promises-safe lib/test/**/*_test.js test/generate_test/lib/**/*_test.js --timeout 100000 --bail --exit",
         "test:circleci": "yarn test:coverage",
         "test:coverage": "nyc npm run test --all && yarn coverage:report:lcov",
-        "coverage:report:lcov": "nyc report --reporter=text-lcov > coverage/lcov.info"
+        "coverage:report:lcov": "nyc report --reporter=text-lcov > coverage/lcov.info",
+        "rebuild_and_test": "run-s build test",
+        "test:profiler": "SOLIDITY_PROFILER=true run-s build run_mocha profiler:report:html",
+        "test:trace": "SOLIDITY_REVERT_TRACE=true run-s build run_mocha",
+        "compile": "sol-compiler",
+        "watch": "sol-compiler -w",
```

Comment:
These commands should be namespaced. These commands would also make sense in the context of the Typescript code in this repo. `compile:sol` and `watch:sol`